synapse/rest/media/v1/thumbnailer.py

```diff
@@ -89,7 +95,8 @@ def transpose(self) -> Tuple[int, int]:
             # Safety: `transpose` takes an int rather than e.g. an IntEnum.
             # self.transpose_method is set above to be a value in
             # EXIF_TRANSPOSE_MAPPINGS, and that only contains correct values.
-            self.image = self.image.transpose(self.transpose_method)  # type: ignore[arg-type]
+            with self.image:
```

Comment:
This will close `self.image` even if the function errors and `self.image` isn't replaced, but I don't think we ever reuse the thumbnailer if these functions error.